src/org/zaproxy/zap/extension/proxies/OptionsLocalProxyPanel.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.proxies;
+
+import java.awt.CardLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.util.List;
+
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.utils.NetworkUtils;
+import org.zaproxy.zap.utils.ZapPortNumberSpinner;
+
+public class OptionsLocalProxyPanel extends JPanel {
+
+    private static final long serialVersionUID = -1350537974139536669L;
+
+    private JPanel panelLocalProxy = null;

```

Comment:
This and following panel could be removed and the components added directly to the panel itself.